dagger/config/config.go

```diff
@@ -0,0 +1,284 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"dagger.io/dagger"
+	"github.com/go-playground/validator/v10"
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/knadh/koanf/parsers/json"
+	"github.com/knadh/koanf/parsers/toml"
+	"github.com/knadh/koanf/parsers/yaml"
+	"github.com/knadh/koanf/providers/file"
+	"github.com/knadh/koanf/v2"
+	"github.com/mitchellh/copystructure"
+)
+
+type BuilderConfig struct {
+	Images   ImagesConfig   `koanf:"images"`
+	Charts   ChartsConfig   `koanf:"charts"`
+	Lint     bool           `koanf:"lint"`
+	Test     bool           `koanf:"test"`
+	Coverage CoverageConfig `koanf:"coverage"`
+}
+
+type ImagesConfig struct {
+	Opni        ImageTarget      `koanf:"opni"`
+	OpniMinimal ImageTarget      `koanf:"minimal"`
+	Opensearch  OpensearchConfig `koanf:"opensearch"`
+	PythonBase  ImageTarget      `koanf:"python-base"`
+	PythonTorch ImageTarget      `koanf:"python-torch"`
+}
+
+type ImageTarget struct {
+	Push           bool       `koanf:"push"`
+	Repo           string     `koanf:"repo" validate:"required_if=Push true,reference-name-only"`
```

Comment:
Currently, the repo config also requires the image name, which is different behaviorally from before where it only required the registry and repo. It feels like it isn't useful to configure image names here, since the differentiating factor should be the tags anyways

Reply:
Technically speaking, the image name _is_ the repo. It's necessary to provide it since it corresponds to the username and password.

Reply:
I was more so wondering if we could keep what we had before with repo=`<reg>/<username>`, because specifying the image name isn't very useful `repo=<reg>/<username>/<name>`, it feels more intuitive that pushing the opni image for example should always be `<reg>/<username>/opni`, with the only differentiating factor being the tag

Or at least provide a sane default for each image in the config iif repo is only set to `<reg>/<username>` .
Alternatively,  have separate fields for registry/username/repo, with sane defaults for repo

Reply:
We could, but the old way ended up being kinda not-obvious, and one of the things I wanted to do with this config system is make everything super obvious. That's why for example you have to specify the repo for each image explicitly instead of having a magic global default or something. The image repo field here is supposed to match the helm-style `image: repository: ...` configuration.

Reply:
Ok maybe in setup config, when we enter the registry can we have options 
> registry
> registry username
> use default image repos ? Y/n
Setting up multiple config files for stuff can become tedious, especially when I think most people will be using the same repos